### PR TITLE
Add a note that QuadSpline is a b-spline and a way to iterate as quads

### PR DESCRIPTION
**Pull request by rsheeter**
Discussion this morning immediately revealed divergent interpretations of the QuadSpline; the b-spline ref should make it clear.

Built on top of #238 to try to keep clippy happy.

### CODE REVIEW

**Review by dfrg — APPROVED** [review]
Looks good to me with suggestions inline. Update the body of the iterator at your discretion :)

File: src/quadspline.rs [hunk 1]
```diff
@@ -31,4 +32,110 @@ impl QuadSpline {
     pub fn points(&self) -> &[Point] {
         &self.0
     }
+
+    /// Return an iterator over the implied [`QuadBez`] sequence.
+    ///
+    /// The returns quads are guaranteed to be G1 continuous.
+    #[inline]
+    pub fn iter(&self) -> impl Iterator<Item = QuadBez> + '_ {
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply src/quadspline.rs:40]: Done

File: src/quadspline.rs [hunk 2]
```diff
@@ -31,4 +32,110 @@ impl QuadSpline {
...
+            self.points[self.idx]
+        } else {
+            self.points[self.idx].midpoint(self.points[self.idx + 1])
+        };
+        let p1 = self.points[self.idx + 1];
+        let p2 = if self.idx + 2 == self.points.len() - 1 {
+            self.points[self.idx + 2]
+        } else {
+            self.points[self.idx + 1].midpoint(self.points[self.idx + 2])
+        };
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply src/quadspline.rs:71]: TIL theres a `get(range)`, very nice.

[the remark this answers is not in the document]
Comment (rsheeter) [reply src/quadspline.rs:71]: Much nicer, so updated.

File: src/quadspline.rs [hunk 3]
```diff
@@ -31,4 +32,110 @@ impl QuadSpline {
     pub fn points(&self) -> &[Point] {
         &self.0
     }
+
+    /// Return an iterator over the implied [`QuadBez`] sequence.
+    ///
+    /// The returns quads are guaranteed to be G1 continuous.
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply src/quadspline.rs:38]: Fixed

**Pull request merged**